### ISSUE TITLE
Sony Alpha-A6000 suffers from several seconds timeout since 2.5.11 (2.5.10 worked)

### ISSUE THREAD

**Issue by MerlijnWajer** [gphoto/libgphoto2#354]
Hi,

While looking to upgrade to a newer Ubuntu and newer gphoto I encountered that gphoto times out for a couple of seconds after image capture and image fetch are already completed. `2.5.10` works fine, `2.5.11` does not, and neither does the latest version (`2.5.22`). Now taking a photo takes quite a bit longer, which is unfortunate, as the software and camera seem to be idle otherwise.

The model is a `Sony Alpha-A6000`. My suspicion is that this commit first introduces the problem: https://github.com/gphoto/libgphoto2/commit/92d0bc4556966ff5419b943c19582e00bc062ef0

Here's a log using 2.5.10: https://wizzup.org/old-gphoto.log
Here's a log using a newer versions - although they seem slightly mixed (but I also verified that it doesn't work on other machines with a clean install): https://wizzup.org/new-gphoto.log

Since I tested with a recent version, I suspect other things might clobber the real issue, but this looks suspicious to me:

```1.813126 ptp_usb_getresp             (2): Reading PTP_OC 0x1009 (Get object) response...
1.813153 gp_port_read                (3): Reading 1024 = 0x400 bytes from port...
1.816131 _cb_irq                     (2): 0x55d129c3a840 with status 0
1.816162 _cb_irq                     (3): interrupt (hexdump of 16 bytes)
0000  10 00 00 00 04 00 02 c2-ff ff ff ff 01 c0 ff ff  ................

1.816189 _cb_irq                     (2): Requeuing completed transfer 0x55d129c3a840
```
`status 0` is `LIBUSB_TRANSFER_COMPLETED`, so why is it requeued?

Followed by timeout-loops, and then:

```
5.131103 _close_async_interrupts     (2): canceling transfer 0:0x55d129c3d9f0 (status 0)
5.131367 _close_async_interrupts     (2): canceling transfer 1:0x55d129c48f70 (status 0)
5.131601 _close_async_interrupts     (2): canceling transfer 2:0x55d129c7ae80 (status 0)
5.131829 _close_async_interrupts     (2): canceling transfer 3:0x55d129c51110 (status 0)
5.132057 _close_async_interrupts     (2): canceling transfer 4:0x55d129c4e1a0 (status 0)
5.132329 _close_async_interrupts     (2): canceling transfer 5:0x55d129c3a840 (status 0)
5.132560 _close_async_interrupts     (2): canceling transfer 6:0x55d129c390d0 (status 0)
5.132793 _close_async_interrupts     (2): canceling transfer 7:0x55d129c4a310 (status 0)
5.133020 _close_async_interrupts     (2): canceling transfer 8:0x55d129c423f0 (status 0)
5.133283 _close_async_interrupts     (2): canceling transfer 9:0x55d129c49c00 (status 0)
5.133741 _cb_irq                     (2): 0x55d129c3d9f0 with status 3
5.133855 _cb_irq                     (2): Transfer 0x55d129c3d9f0 should be in LIBUSB_TRANSFER_COMPLETED, but is 3!
5.134057 _cb_irq                     (2): 0x55d129c48f70 with status 3
5.134166 _cb_irq                     (2): Transfer 0x55d129c48f70 should be in LIBUSB_TRANSFER_COMPLETED, but is 3!
5.134360 _cb_irq                     (2): 0x55d129c7ae80 with status 3
5.134469 _cb_irq                     (2): Transfer 0x55d129c7ae80 should be in LIBUSB_TRANSFER_COMPLETED, but is 3!
5.134666 _cb_irq                     (2): 0x55d129c51110 with status 3
5.134775 _cb_irq                     (2): Transfer 0x55d129c51110 should be in LIBUSB_TRANSFER_COMPLETED, but is 3!
5.134971 _cb_irq                     (2): 0x55d129c4e1a0 with status 3
5.135087 _cb_irq                     (2): Transfer 0x55d129c4e1a0 should be in LIBUSB_TRANSFER_COMPLETED, but is 3!
5.135283 _cb_irq                     (2): 0x55d129c3a840 with status 3
5.135394 _cb_irq                     (2): Transfer 0x55d129c3a840 should be in LIBUSB_TRANSFER_COMPLETED, but is 3!
5.135594 _cb_irq                     (2): 0x55d129c390d0 with status 3
5.135725 _cb_irq                     (2): Transfer 0x55d129c390d0 should be in LIBUSB_TRANSFER_COMPLETED, but is 3!
5.135892 _cb_irq                     (2): 0x55d129c4a310 with status 3
5.135985 _cb_irq                     (2): Transfer 0x55d129c4a310 should be in LIBUSB_TRANSFER_COMPLETED, but is 3!
5.136150 _cb_irq                     (2): 0x55d129c423f0 with status 3
5.136242 _cb_irq                     (2): Transfer 0x55d129c423f0 should be in LIBUSB_TRANSFER_COMPLETED, but is 3!
5.136507 _cb_irq                     (2): 0x55d129c49c00 with status 3
5.136654 _cb_irq                     (2): Transfer 0x55d129c49c00 should be in LIBUSB_TRANSFER_COMPLETED, but is 3!
```

3 seems to be `LIBUSB_TRANSFER_TIMED_OUT`.

**Comment by msmeissn** [comment]
gphoto2 is waiting for 3 additional seconds for either CAPTURECOMPLETE or the secondary raw image.
(this is a hack to allow RAW+JPG capture ... :/ )
I will see if I can think of something to shorten it. :/

**Comment by MerlijnWajer** [comment]
Maybe there could be an option to disable/enable the extra wait?

**Comment by MerlijnWajer** [comment]
Could you point me to the code that does this, so I can try to see if that helps?

**Comment by MerlijnWajer** [comment]
I did a random test and changed the timeout from 3000 to 100 in `gphoto2`, and that seemed to work. Need to check if the images are still fine, though.

**Comment by msmeissn** [comment]
this was the right place :)

**Comment by msmeissn** [comment]
(and the requeuing is just we have 10 buffers for interrupts and reuse them aftre receiving interrupts, so harmless)

**Comment by MerlijnWajer** [comment]
I will be able to test this somewhere at the end of this month earliest - I don't have access to the hardware normally. Hope that is OK.

**Comment by msmeissn** [comment]
I added capturecomplete synthesizing, which should now skip the 3 second wait

**Comment by MerlijnWajer** [comment]
Alright, I wasn't sure if you needed me to test it before closing the issue. Thanks for fixing it! I will test when I can.

**Comment by msmeissn** [comment]
I found the time to borrow a Sony A7R2 and tested it ;) But please report if it also fixed it for you when you get to it!

**Comment by roycamp** [comment]
Hi @msmeissn, I believe this fix significantly improved the performance of the Sony RX0.  It was taking 8 seconds for RAW+JPEG and 6 seconds for RAW. After the fix it is 4 seconds and 2.5 seconds.  These times still seem high though.  Here is the log after the fix for RAW:
https://pastebin.com/inyhaTb4
and RAW+JPEG:
https://pastebin.com/q9A6xV35

Thoughts? Thanks!

**Comment by msmeissn** [comment]
There seems to be some 0.2 seconds or so that we might be able to shave off at the end, but otherwise it looks as fast as we can get it
